Rpackage/R/fitAndScore.R

```diff
@@ -0,0 +1,191 @@
+fit_and_score <- structure(function(tree, target.mat, feature.mat, 
+                                    parameters, feature_names=NULL, 
+                                    n_folds = 3, scorer=NULL, loss = "hinge",
+                                    pruning = TRUE){
+  ### total length
+  l <- length(target.mat[, 1])
```

Comment:
Minor comment: it's probably more efficient to use l <- dim(target.mat)[1]. The dimensions are probably stored in the data structure and accessible in O(1) time, whereas your current approach performs a slice before checking the length.

Reply:
Actually, you can use the nrow function for that.